### PR TITLE
Update quantile suffix function

### PR DESCRIPTION
**Pull request by cw9**
Updated the quantile function to generate ".p9999" rather than ".p99.99" for P9999

@xichen2020

### CODE REVIEW

File: aggregator/options.go [hunk 1]
```diff
@@ -796,7 +797,11 @@ func (o *options) computeDefaultSuffixes() {
 		default:
 			q, ok := aggType.Quantile()
 			if ok {
-				o.defaultSuffixes[aggType.ID()] = o.timerQuantileSuffixFn(q)
+				b, err := o.timerQuantileSuffixFn(q)
```
Comment (xichen2020) [inline aggregator/options.go]: Perhaps it'll be better to still have the function return a `[]byte` and panic inside the function on error instead?

File: aggregator/options.go [hunk 2]
```diff
@@ -869,8 +874,20 @@ func (o *options) computeFullGaugePrefix() {
...
 // By default we use e.g. ".p50", ".p95", ".p99" for the 50th/95th/99th percentile.
-func defaultTimerQuantileSuffixFn(quantile float64) []byte {
-	return []byte(".p" + strconv.FormatFloat(quantile*100, 'f', -1, 64))
+func defaultTimerQuantileSuffixFn(quantile float64) ([]byte, error) {
+	if quantile < 0 || quantile > 1 {
+		return nil, fmt.Errorf("invalid quantile %f", quantile)
+	}
+	str := strconv.FormatFloat(quantile*100, 'f', -1, 64)
+
+	if strings.Contains(str, ".") {
```
Comment (xichen2020) [inline aggregator/options.go]: nit: Can directly call `strings.Index(str, ".")` here and check it against -1.

File: aggregator/options.go [hunk 3]
```diff
@@ -869,8 +874,20 @@ func (o *options) computeFullGaugePrefix() {
 }
 
 // By default we use e.g. ".p50", ".p95", ".p99" for the 50th/95th/99th percentile.
-func defaultTimerQuantileSuffixFn(quantile float64) []byte {
-	return []byte(".p" + strconv.FormatFloat(quantile*100, 'f', -1, 64))
+func defaultTimerQuantileSuffixFn(quantile float64) ([]byte, error) {
+	if quantile < 0 || quantile > 1 {
```
Comment (xichen2020) [inline aggregator/options.go]: Probably don't need this? Alternatively this should be validated when setting the quantiles?

File: aggregator/options.go [hunk 4]
```diff
@@ -869,8 +874,20 @@ func (o *options) computeFullGaugePrefix() {
...
+		return nil, fmt.Errorf("invalid quantile %f", quantile)
+	}
+	str := strconv.FormatFloat(quantile*100, 'f', -1, 64)
+
+	if strings.Contains(str, ".") {
+		if quantile <= 0.99 {
```
Comment (xichen2020) [inline aggregator/options.go]: Perhaps we just always remove the dot regardless?

Comment (cw9) [reply aggregator/options.go]: Yeah, I'm ok either way, gonna remove the error/panic and just remove the dot, we have control over the valid aggregation types list anyway.

File: aggregator/options.go (hunk 2 above)
Comment (cw9) [reply aggregator/options.go]: sg

File: aggregator/options.go [hunk 5]
```diff
@@ -870,7 +870,13 @@ func (o *options) computeFullGaugePrefix() {
 
 // By default we use e.g. ".p50", ".p95", ".p99" for the 50th/95th/99th percentile.
 func defaultTimerQuantileSuffixFn(quantile float64) []byte {
-	return []byte(".p" + strconv.FormatFloat(quantile*100, 'f', -1, 64))
+	str := strconv.FormatFloat(quantile*100, 'f', -1, 64)
+
```
Comment (xichen2020) [inline aggregator/options.go]: nit: remove the superfluous blank line.

**Pull request merged**